sycl/source/detail/scheduler/graph_builder.cpp

```diff
@@ -232,16 +232,29 @@ Command *Scheduler::GraphBuilder::insertMemoryMove(MemObjRecord *Record,
 
   AllocaCommandBase *AllocaCmdSrc =
       findAllocaForReq(Record, Req, Record->MCurContext);
+  if (IsSuitableSubReq(Req) && !AllocaCmdSrc) {
+    auto IsSuitableAlloca = [Record, Req](AllocaCommandBase *AllocaCmd) {
+      bool Res = sameCtx(AllocaCmd->getQueue()->get_context_impl(),
```

Comment:
Cannot imagine code which would trigger such a situation, but what if the first alloca with the same context is not full alloca but alloca for sub buffer which doesn't match our requirement?

Reply:
> Cannot imagine code which would trigger such a situation

Case which would trigger this code was added as a test case.
> what if the first alloca with the same context is not full alloca but alloca for sub buffer which doesn't match our requirement

Thanks for pointing that out! Added additional check.

Reply:
No, I meant that if the vector has allocation for subbuffer before allocation for the whole buffer for the given context, the subbuffer will be chosen.

Reply:
Oh, you are making sure that the alloca is a "parent" one by checking the type. Please, disregard my previous comment.
But I think check for memory object is redundant because all alloca commands in the vector should have "correct" mem obj otherwise there is a bug somewhere, suggest removing the check.

Reply:
And could you please add the comment explaining why you check for the type of alloca.

Reply:
Wrote comments and removed redundant check